src/App.jsx

```diff
@@ -14,33 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ABC = 'Sort alphabetically';
+const SORT_FIELD_LENGTH = 'Sort by length';
+const SORT_FIELD_RESET = 'Reset';
+const SORT_FIELD_REVERSE = 'Reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [currentSort, setCurrentSort] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  let sortedGoods = [...goodsFromServer];
+  const isReset = currentSort || reversed;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const handleReset = () => {
+    setCurrentSort('');
+    setReversed(false);
+  };
+
+  switch (currentSort) {
+    case SORT_FIELD_ABC:
+      sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SORT_FIELD_LENGTH:
+      sortedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    case SORT_FIELD_RESET:
+      sortedGoods = [...goodsFromServer];
+      break;
+    default:
+      break;
+  }
+
+  if (reversed) {
+    sortedGoods = sortedGoods.toReversed();
```

Comment:
Create a function that will keep the sort logic, and move it outside the component.